manticore/core/state.py

```diff
@@ -111,7 +111,7 @@ def __enter__(self):
         assert self._child is None
         new_state = State(self._constraints.__enter__(), self._platform)
         new_state._input_symbols = list(self._input_symbols)
-        new_state._context = copy.deepcopy(self._context)
+        new_state._context = copy.copy(self._context)
```

Comment:
what effect does this have?

Reply:
This is probably wrong. We want to deepcopy everything (list/dict/etc) except the Expressions saved in the context. Expressions should be copied just by reference.
So there are no 2 Variables with the same name.